### PR TITLE
Move unnecessary_string_escapes tests to reflective

### PR DESCRIPTION
**Pull request by srawlins**

### CODE REVIEW

**Review by pq — CHANGES_REQUESTED** [review]
I'm not seeing the new reflective tests... Maybe they didn't get committed?

**Comment by srawlins** [comment]
Lol thanks Phil! Yes I did not `git add` them appropriately. Done!

File: test/rules/unnecessary_string_escapes_test.dart [hunk 1]
```diff
@@ -0,0 +1,210 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
```
Comment (pq) [inline test/rules/unnecessary_string_escapes_test.dart:1]: I think this should be 2023?  (Or 2020 if you've been keeping the copyright from the migrated tests?)

Comment (srawlins) [reply test/rules/unnecessary_string_escapes_test.dart]: done!

**Review by pq — APPROVED** [review]
💎

**Pull request merged**